Automattic/wp-calypso
# Marketplace Themes: Adds a loading state while the site is transferred to atomic

#### Proposed Changes

* Adds a loading state while the site is transferred to atomic on the theme page.

https://user-images.githubusercontent.com/1234758/216349115-bef9673b-881a-4ba9-8a17-dae21b440fa2.mov

* I'm using the `/rest/v1.1/sites/{SITE_ID}/themes/mine` endpoint to check if the active theme is the recently subscribed theme.

#### Testing Instructions

* Create a free site
* Navigate to `/theme/makoney/{YOUR_FREE_SITE_SLUG}`
* Click on the CTA `Upgrade to subscribe` and go through the checkout flow.
* Once you land on the theme page again, you should see a loading state while the site is being transferred.

Slack discussion: p1675165572273799-slack-C9EJ7KSGH
